audformat/core/utils.py

```diff
@@ -550,17 +558,14 @@ def intersect(
     objs = _maybe_convert_single_level_multi_index(objs)
     _assert_index_alike(objs)
 
-    # sort objects by length
-    objs = sorted(objs, key=lambda obj: len(obj))
-    # start from shortest index
-    index = objs[0]
-    mask = np.ones(len(index), dtype=bool)
+    # start from first object
+    index = list(objs[0])
     for obj in objs[1:]:
-        mask &= index.isin(obj)
-        if not mask.any():
+        index = [idx for idx in index if idx in obj]
```

Comment:
Really surprising this if faster than using `.isn()`.

Reply:
Yes, and a lot if you compare 7.81 s with 0.08 s

Reply:
And it also seems strange to me that in some situation it is faster than using `set`. Did you also test `set` with sorting?

Reply:
I don't know, maybe my implementation is wrong?
I think converting it to a set takes already time.
Here is how I would implement it in the current code:

```python
    # sort objects by length
    objs_sorted = sorted(objs, key=lambda obj: len(obj))
    # start from first object
    index = set(objs_sorted[0])
    for obj in objs_sorted[1:]:
        index = index.intersection(set(obj))
        if len(index) == 0:
            # break early if no more intersection is possible
            break

    index = _alike_index(objs[0], list(index))
```
and here the results compared to the current pull request

| Table type | Index samples | set | pull request |
| --- | --- | --- | --- |
| filewise | (100000, 500000, 100000) | 0.29 s | 0.16 s |
| filewise | (100000, 500000, 10000) | 0.23 s | 0.07 s |
| segmented | (100000, 500000, 100000) | 6.89 s | 2.71 s |
| segmented | (100000, 500000, 10000) | 6.02 s | 0.27 s |

Reply:
The disadvantage with using `set()` might be that we have to convert every object to a set, also the very long indices (here the one with 500,000 samples). Whereas with `list()` we just convert the shortest index to a list and then check for all entries in that list if they are part of the very long index, which might be faster.

Reply:
Yes, that sounds reasonable. Results look very good now!